src/d3d9/d3d9_device.cpp

```diff
@@ -5771,7 +5766,7 @@ namespace dxvk {
       GenerateTextureMips(texturesToGen);
 
     auto* ibo = GetCommonBuffer(m_state.indices);
-    if (ibo != nullptr && ibo->NeedsUpload())
+    if (ibo != nullptr && ibo->Desc()->Pool != D3DPOOL_DEFAULT  && !ibo->DirtyRange().IsDegenerate())
```

Comment:
Maybe add a NeedsUpload helper method here. Feel free to pick a better name. Maybe NeedsDeferredUpload